src/session.rs

```diff
@@ -718,7 +767,7 @@ where
                     .await?;
 
                 Ok(Client {
-                    tx,
+                    tx: client.tx,
                     is_legacy: true,
                     new_session_response: None,
                 })
```

Comment:
Similarly here I think you can set `client.is_legacy = true` and return `client`.